gateway-ha/src/main/java/io/trino/gateway/ha/router/RESTApiRoutingGroupSelector.java

```diff
@@ -0,0 +1,198 @@
+        }
+        this.uri = uri;
+        this.httpClient = new JettyHttpClient(new HttpClientConfig());
+    }
+
+    @Override
+    public String findRoutingGroup(HttpServletRequest servletRequest)
+    {
+        if (uri == null) {
+            log.warn("URI is null, cannot send REST API request for routing group.");
+            return servletRequest.getHeader(ROUTING_GROUP_HEADER);
+        }
+
+        try {
+            String requestBody = createRequestBody(servletRequest);
+            Request.Builder requestBuilder = new Request.Builder()
+                    .setUri(uri)
+                    .setMethod("POST")
+                    .setBodyGenerator(createStaticBodyGenerator(requestBody, UTF_8));
+            requestBuilder.addHeaders(getValidHeaders(servletRequest));
+            // set Content-Length for updated request body
+            requestBuilder.setHeader("Content-Length", String.valueOf(requestBody.getBytes(UTF_8).length));
+
+            Request request = requestBuilder.build();
+            StringResponseHandler.StringResponse response = httpClient.execute(request, createStringResponseHandler());
+            if (response.getStatusCode() != 200) {
+                throw new RuntimeException("Unexpected response code " + response.getStatusCode());
+            }
+            return extractRoutingGroup(response.getBody());
+        }
+        catch (Exception e) {
+            log.error(e, "Error occurred while retrieving routing group "
+                    + "from Routing Rules REST API at " + uri);
+        }
+        return servletRequest.getHeader(ROUTING_GROUP_HEADER);
+    }
+
+    private String createRequestBody(HttpServletRequest request)
+            throws JsonProcessingException
+    {
+        Map<String, Object> combinedMap = new HashMap<>();
+        combinedMap.put("Content-Type", "application/json");
+
+        // Convert the map to a JSON string
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+        objectMapper.registerModule(new Jdk8Module());
+
+        if (requestAnalyzerConfig.isAnalyzeRequest()) {
+            TrinoQueryProperties trinoQueryProperties = new TrinoQueryProperties(request, requestAnalyzerConfig);
+            TrinoRequestUser trinoRequestUser = trinoRequestUserProvider.getInstance(request);
+
+            String trinoQueryPropertiesJson = objectMapper.writeValueAsString(trinoQueryProperties);
+            String trinoRequestUserJson = objectMapper.writeValueAsString(trinoRequestUser);
+
+            combinedMap.put("trinoQueryProperties", objectMapper.readValue(trinoQueryPropertiesJson, Map.class));
+            combinedMap.put("trinoRequestUser", objectMapper.readValue(trinoRequestUserJson, Map.class));
+        }
+
+        combinedMap.putAll(extractHttpDetails(request));
+        return objectMapper.writeValueAsString(combinedMap);
+    }
+
+    private String extractRoutingGroup(String responseBody)
```

Comment:
Similarly create a class to represent the response using `@JsonProperty` annotation and an airlift `ResponseHandler` to deserialize. `ClusterStatsInfoApiMonitor` gives a simple example of this